### ISSUE TITLE
withNamespace HOC - Namespaces not loaded on changeLanguage

### ISSUE THREAD

**Issue by Mucht** [i18next/react-i18next#1020]
**Describe the bug**
`key "xxx" for namespace "yyy" won't get resolved as namespace was not yet loaded`
This error pop on every single key of my project if I try to change the language with `i18next.changeLanguage("en-US")`.
I think my config is ok since everything is fine before I call `changeLanguage`.

**Occurs in react-i18next version**
react-i18next@^9.0.10

**To Reproduce**
I'm using `withNamespace HOC`

**Expected behaviour**
The same behaviour than when I load the page without changing the language.

**OS (please complete the following information):**

- Device: all
- Browser: all

**Additional context**
```
i18next
    .use(i18next-xhr-backend)
    .use(i18next-browser-languagedetector)
    .on("languageChanged", lng => {
        moment.locale(lng);
    })
    .init({
        debug: true,
        ns: [],
        defaultNS: "",
        load: "languageOnly",
        keySeparator: ".",
        fallbackLng: "fr",
        interpolation: {
            escapeValue: false, // react already escapes value
        },
        react: {
            wait: true,
            nsMode: "fallback", // default = "default", read below
        },
        backend: {
            addPath: `${root("static")}/translations/add/{{lng}}/{{ns}}.json`,
            loadPath: `${root("static")}/translations/{{lng}}/{{ns}}.json`,
        },
    });
```

**Comment by jamuhl** [comment]
for react-i18next < v10 you might keep i18next fixed below https://github.com/i18next/i18next/blob/master/CHANGELOG.md#1710 (< v17.1.0) or consider upgrading react-i18next

**Comment by Mucht** [comment]
Below 17.1.0 didn't fix anything as I expected since it just removed the warning and didn't solve the issue.

But it actually fixed it on v19.02...

Thanks for your time 👍🏼
